lib/bsearch.c

```diff
@@ -14,7 +14,7 @@
  *
  */
 
-#include HAL
+#include "../hal/hal.h"
```

Comment:
please note that You have `-I.` in `CFLAGS` to avoid the need to use `..` across includes (which for some reason is ignored in many places :)). 

Side note: I think that `-I.` is also unneeded as CWD is always added to local include path - otherwise `include "file.h" would not work (alas might have change how `include <file.h>` works).

Reply:
I agree with You that this part also should be cleaned up, but like I wrote in a description that there are coming changes in headers in `phoenix-rtos-kernel` so I have decided to wait on it before making a bigger changes.